src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -30,12 +32,55 @@
  * if that's what you want to do.
  */
 public final class BwaMemIndex implements AutoCloseable {
+
+    /**
+     * Indexing algorithms supported by Bwa.
+     */
+    public enum Algorithm {
+        RB2, IS, AUTO;
+
+        /**
+         * The string name use by the Bwa command line to denote that algorithm.
+         * @return never {@code null} and unique across algorithms.
+         */
+        public String toBwaName() {
+            return this.toString().toLowerCase();
+        }
+    }
+
     private final String indexImageFile; // stash this for error messages
     private volatile long indexAddress; // address where the index was memory-mapped (for use by C code)
     private final AtomicInteger refCount; // keep track of how many threads are actively aligning
     private final List<String> refContigNames; // the reference dictionary from the index
     private static volatile boolean nativeLibLoaded = false; // whether we've loaded the native library or not
 
+    private static final List<String> FASTA_EXTENSIONS = Collections.unmodifiableList(Arrays.asList(".fasta", ".fa"));
+
+
+    public static void indexReference(final String referenceName, final String indexPrefix, final Algorithm algo) {
+        final String referenceFileName = resolveReferenceFileName(referenceName);
+        loadNativeLibrary();
+        createReferenceIndex(referenceFileName, indexPrefix, algo.toBwaName());
```

Comment:
I think you two comment refer to two different things... 

@magicDGS mmmm.... perhaps makes more sense to provide a function that check whether the index files seem to be present and the invoking code may decide whether to proceed.

@tedsharpe I think this may become a bit more complex discussion and we would need to think of what are the use case we may want to cover.... 

for example are you suggesting that if we are missing only the image file we should discard the index files? How long it would take to index the human genome (~3.2Gb).  Perhaps there is bwa version stamp in the index files that we could check on?